src/azul/__init__.py

```diff
@@ -991,23 +1006,40 @@ def _git_status(self) -> Mapping[str, str]:
         }
 
     @property
-    def lambda_git_status(self) -> Mapping[str, str]:
+    def lambda_git_status(self) -> dict[str, str]:
         return {
             'commit': self.environ['azul_git_commit'],
             'dirty': str_to_bool(self.environ['azul_git_dirty'])
         }
 
+    @property
+    def _aws_account(self) -> dict[str, str]:
+        return {
+            'azul_aws_account_name': self.aws_account_name
+        }
+
+    @property
+    def aws_account_name(self) -> str:
+        try:
+            return self.environ['azul_aws_account_name']
+        except KeyError:
+            from azul.deployment import (
```

Comment:
As you pointed out, we can't use `config.lamba_context` here. What we should be able to use is the presence of any variable documented here:

https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html#configuration-envvars-runtime

I think `AWS_LAMBDA_FUNCTION_NAME` makes the most sense. A `config` method that returns `True` if that variable is set should be added and used here to distinguish the benign case—in which the fall back is used—from the error case—in which the `KeyError` is reraised.